eclipse/src/saros/editor/EditorManager.java

```diff
@@ -1789,4 +1810,25 @@ private void checkThreadAccess() {
   public IActivityProducer getActivityProducer() {
     return activityDelayer;
   }
+
+  /**
+   * Converts the given file delegate to a Saros file object.
+   *
+   * @param fileDelegate the file delegate to convert
+   * @return a Saros file object representing the given file delegate or <code>null</code> if there
+   *     is not running session or the delegate does not belong to a shared reference point
+   * @throws NullPointerException if the given file delegate is <code>null</code>
+   * @see ResourceConverter#convertToFile(Set, IFile)
+   */
+  private saros.filesystem.IFile convertToFile(IFile fileDelegate) {
```

Comment:
I have seen the logic 3 times in the whole patch set. Any reason not to move the logic to the ResourceConverter and adding (a) method(s) that accepts a session object and an IResource (IFile, IFolder) object ?

Reply:
All this method does is check whether the given session object is `null`, request the currently shared reference points and call `ResourceConverter.convertToFile(...)` with it.

In most other cases, the check whether the session object is valid is already done beforehand by different logic. In general, I don't think it is necessary to introduce a separate method for this in the `ResourceConverter`.

Regardless, this is a very minor change that can also be done after the fact, so I don't think it should hold back this PR.